test/optional-validation.test.js

```diff
@@ -0,0 +1,595 @@
+    User.create({"name": "Groover"}, function(err, user){
+      if (err) return cb(err);
+      data = {name: name};
+      data.id = user.id;
+      User.updateOrCreate(data, options, cb);
+    });
+  }
+
+  function getNewWhere() {
+    return {name: 'DoesNotExist' + (whereCount++)};
+  }
+
+  describe('no model setting', function () {
+
+    describe('method create', function() {
+      it('should throw on create with validate:true with invalid data', function (done) {
+        User.create(INVALID_DATA, {validate: true}, expectValidationError(done));
+      });
+
+      it('should NOT throw on create with validate:false with invalid data', function (done) {
+        User.create(INVALID_DATA, {validate: false}, expectCreateSuccess(INVALID_DATA, done));
+      });
+
+      it('should NOT throw on create with validate:true with valid data', function (done) {
+        User.create(VALID_DATA, {validate: true}, expectCreateSuccess(done));
+      });
+
+      it('should NOT throw on create with validate:false with valid data', function (done) {
+        User.create(VALID_DATA, {validate: false}, expectCreateSuccess(done));
+      });
+
+      it('should throw on create with invalid data', function (done) {
+        User.create(INVALID_DATA, expectValidationError(done));
+      });
+
+      it('should NOT throw on create with valid data', function (done) {
+        User.create(VALID_DATA, expectCreateSuccess(done));
+      });
+    });
+
+    describe('method findOrCreate', function() {
+      it('should throw on findOrCreate with validate:true with invalid data', function (done) {
+        User.findOrCreate(getNewWhere(), INVALID_DATA, {validate: true}, expectValidationError(done));
+      });
+
+      it('should NOT throw on findOrCreate with validate:false with invalid data', function (done) {
+        User.findOrCreate(getNewWhere(), INVALID_DATA, {validate: false}, expectCreateSuccess(INVALID_DATA, done));
+      });
+
+      it('should NOT throw on findOrCreate with validate:true with valid data', function (done) {
+        User.findOrCreate(getNewWhere(), VALID_DATA, {validate: true}, expectCreateSuccess(done));
+      });
+
+      it('should NOT throw on findOrCreate with validate:false with valid data', function (done) {
+        User.findOrCreate(getNewWhere(), VALID_DATA, {validate: false}, expectCreateSuccess(done));
+      });
+
+      it('should throw on findOrCreate with invalid data', function (done) {
+        User.findOrCreate(getNewWhere(), INVALID_DATA, expectValidationError(done));
+      });
+
+      it('should NOT throw on findOrCreate with valid data', function (done) {
+        User.findOrCreate(getNewWhere(), VALID_DATA, expectCreateSuccess(done));
+      });
+    });
+
+    describe('method updateOrCreate with new data', function() {
+      it('should throw on updateOrCreate with validate:true with invalid data', function (done) {
+        User.updateOrCreate(INVALID_DATA, {validate: true}, expectValidationError(done));
```

Comment:
IMHO it's not a false assumption, because it does what is says. It takes new data and uses updateOrCreate. If in the future the short-circuit would be changed to something else, there would be no tests here to check if updateOrCreate uses the optionalValidation in the intended way. Does that makes sense?

But if you think it's better, I'll remove it! Just let me know!